src/lib.rs

```diff
@@ -147,7 +154,7 @@ pub trait Actor: 'static + Send + Sized {
 
     /// Called as soon as the actor has been started.
     #[allow(unused_variables)]
-    async fn started(&mut self, ctx: &mut Context<Self>) {}
+    async fn started(&mut self, mailbox: &mut Mailbox<Self>) {}
```

Comment:
Not entirely sure about this change. It feels a bit weird to pass the `Mailbox` but the upside is that people can no longer call `stop` in here which simplifies the event loop.

Reply:
Would it be possible to just pass the address in?

Reply:
If you just pass the address in, then the user can't call functions like `xtra::join` or `xtra::select` in `started` which is a regression in terms of features.

I think it could be convinced that it is the right thing to do.

Reply:
It could be quite interesting to have Context have a second argument, representing the life cycle stage. So, handle would take Context<A, Running> and started would take Context<A, Starting>. Stop would only exist for Context<A, Running>. On the other hand, not being able to call stop in started is technically a feature regression too 🤷‍♂️

Reply:
I'd have `started` return `Result<(), Self::Stop>` so you can still stop the actor in `started`, just not via calling `stop`.

A custom event loop would get a warning from the `must_use` on `Result` so they are hinted towards handling that.

We can certainly do the life-cycle type parameter thing, I'll have a play to see how it turns out! I think the advantage would be that all functions just take a `Context`, including `started`.

It will make custom event loops a bit more verbose because users have to construct a `Context` just for being able to call `started`.

Reply:
> A custom event loop would get a warning from the must_use on Result so they are hinted towards handling that.

This is indeed a nice advantage! It might be worth the asymmetry not being able to use Context::stop here like elsewhere (I wonder if this is worth worrying about. I might be a bit confused here as a user)